File: sankey diagram all included.R
# load the required packages
library(networkD3)
library(tidyverse)

# read the data from the CSV file
sankey_data <- read.csv("C:/Users/abhis/PycharmProjects/outreachy/task 5/output.csv")

# create a data frame for the Sankey plot
sankey_plot_data <- sankey_data %>%
  select(from = translations_from, to = translations_to, value = article_count)

# create the nodes data frame
nodes <- data.frame(name = unique(c(sankey_plot_data$from, sankey_plot_data$to)))
nodes$ID <- seq_len(nrow(nodes))

# create the Sankey plot
sankeyPlot <- sankeyNetwork(Links = sankey_plot_data, 
                            Nodes = nodes,
                            Source = "from",
                            Target = "to", 
                            Value = "value",
                            NodeID = "name",
                            sinksRight = FALSE)

# display the Sankey plot
sankeyPlot

File: sankey_diagram_all_included.py
import pandas as pd
import plotly.graph_objects as go

# Load data from CSV file
data = pd.read_csv("output.csv")

# Define source, target, and value columns
source = data["translations_from"]
target = data["translations_to"]
value = data["article_count"]

# Create Sankey diagram figure
fig = go.Figure(data=[go.Sankey(
    node=dict(
        pad=15,
        thickness=20,
        line=dict(color="black", width=0.5),
        label=source.unique().tolist() + target.unique().tolist(),
        color="blue"
    ),
    link=dict(
        source=source.map(lambda x: source.unique().tolist().index(x)),
        target=target.map(lambda x: target.unique().tolist().index(x)),
        value=value,
    ))])

# Show figure
fig.show()
